File: Programmers/Level2/프린터.py
def solution(priorities, location):

    answer = 0
    val = priorities[location]
    priorities[location] = 0

    while priorities:
        temp = priorities.pop(0)
        if priorities:
            if temp == 0:
                if val >= max(priorities):
                    answer += 1 
                    return answer
                else:
                    priorities.append(temp)
            else:
                if temp >= val and temp >= max(priorities) :
                    answer += 1
                else:
                    priorities.append(temp)
        else:
            return answer+1
    

priorities = [1, 1, 9, 1, 1, 1]
location = 0
print(solution(priorities, location))

File: Programmers/Level2/프린터/프린터.js
function solution(priorities, location) {
  let target = location;
  let answer = 1;
  let temp = 0;

  while (priorities.length > 0) {
    temp = priorities.shift();
    if (priorities.some((val) => val > temp)) {
      priorities.push(temp);
    } else {
      if (target === 0) {
        break;
      } else answer++;
    }
    if (target === 0) target = priorities.length - 1;
    else target--;
  }
  return answer;
}

const priorities = [1, 1, 9, 1, 1, 1];
const location = 0;
console.log(solution(priorities, location));
